src/backends/CMakeLists.txt

```diff
@@ -1,50 +1,42 @@
-###############################################################################
-#
-# This file is part of CMake configuration for SOCI library
-#
-# Copyright (C) 2010 Mateusz Loskot
-# Distributed under the Boost Software License, Version 1.0.
-# (See accompanying file LICENSE_1_0.txt or copy at
-# http://www.boost.org/LICENSE_1_0.txt)
-#
-###############################################################################
-colormsg(_HIBLUE_ "Configuring SOCI backend libraries:")
+set(SOCI_EMPTY ${PROJECT_IS_TOP_LEVEL} CACHE STRING "Include the 'empty' backend. Can be bool-valued or one of 'Enabled', 'Disabled' and 'AsAvailable'")
```

Comment:
Does `AsAvailable` really make sense here?

Reply:
I'm open for better suggestions. This is effectively the current behavior where it would just check whether the dependencies are available and if not, disable the backend. This is annoying if you explicitly enabled that backend because you need it (as the cmake invocation will work fine, even though required dependencies are met).